src/transformers/models/beit3/modeling_beit3.py

```diff
@@ -0,0 +1,1447 @@
+
+            encoder_layer_outputs = layer(
+                hidden_state,
+                attention_mask=attention_mask if past_key_value is None else None,
+                image_text_mask=image_text_mask,
+                multiway_split_position=multiway_split_position,
+                past_key_value=past_key_value[idx] if past_key_value is not None else None,
+                output_attentions=output_attentions,
+            )
+            if output_attentions:
+                hidden_state, attention_weights = encoder_layer_outputs
+                all_self_attentions = all_self_attentions + (attention_weights,)
+            else:
+                hidden_state = encoder_layer_outputs
+
+        if output_hidden_states:
+            all_hidden_states = all_hidden_states + (hidden_state,)
+
+        if self.fc_norm is not None:
+            hidden_state = self.fc_norm(hidden_state, split_position=multiway_split_position)
+
+        if not return_dict:
+            return tuple(
+                v
+                for v in [
+                    hidden_state,
+                    all_hidden_states,
+                    all_self_attentions,
+                ]
+                if v is not None
+            )
+
+        return BaseModelOutput(
+            last_hidden_state=hidden_state,
+            hidden_states=all_hidden_states,
+            attentions=all_self_attentions,
+        )
+
+
+@add_start_docstrings(
+    """BEiT-3 is a general-purpose multimodal foundation model that excels in both vision and vision-language tasks. It
+        utilizes [Multiway transformers] (https://arxiv.org/abs/2208.10442) for deep fusion and modality-specific
+        encoding, and unifies masked modeling on images, texts, and image-text pairs, achieving top performance on
+        multiple benchmarks.""",
+    BEIT3_START_DOCSTRING,
+)
+class Beit3Model(Beit3PreTrainedModel):
+    def __init__(self, config, add_pooling_layer=False):
+        super().__init__(config)
+
+        self.text_embedding = nn.Embedding(config.vocab_size, config.hidden_size)
+        self.vision_embedding = Beit3VisionEmbedding(config)
+        self.encoder = Beit3Encoder(config)
+
+        self.pooler = Beit3Pooler(config) if add_pooling_layer else None
+
+        self.post_init()
+
+    def get_input_embeddings(self):
+        return self.text_embedding
+
+    def set_input_embeddings(self, value: nn.Embedding) -> None:
+        self.text_embedding = value
+
+    def get_num_layers(self):
+        return self.encoder.num_layers
+
+    @add_start_docstrings_to_model_forward(BEIT3_MODEL)
+    @replace_return_docstrings(output_type=BaseModelOutputWithPooling, config_class=_CONFIG_FOR_DOC)
+    def forward(
+            self,
+            input_ids=None,
+            pixel_values=None,
+            attention_mask=None,
+            image_text_mask=None,
+            vision_masked_position=None,
+            past_key_value=None,
+            text_end_positions=None,
```

Comment:
this can be inferred from the `input_ids` and `pixel_values`, would be great to remove this argument